fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/OCFLPersistentStorageUtilsTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl.impl;
+
+import org.junit.Test;
+
+import static org.fcrepo.persistence.ocfl.impl.OCFLPersistentStorageUtils.relativizeSubpath;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test class for {@link OCFLPersistentStorageUtils}
+ *
+ * @author dbernstein
+ */
+public class OCFLPersistentStorageUtilsTest {
+
+    @Test
+    public void testRelativizeSubpath() {
```

Comment:
Add a test where `rootObjectId` is the same as `resourceId`.
And maybe one where there is stuff after the root id other than a `/` if that's a valid case?

Reply:
Test added for root equals resource.
I'm not sure I understand the second issue you're raising.  Are you talking about something other than nested subpaths?

Reply:
The second case is related to the second half of the conditional here:
https://github.com/fcrepo4/fcrepo4/blob/fcfea26ded89335cfc1802a1994b1792eac96d9a/fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/OCFLPersistentStorageUtils.java#L97

I think i misread the conditional yesterday though, on second look i'm realizing it is disallowing the case where there is no slash after the root id, which you already have a test for.